kernel/memory_structs/src/lib.rs

```diff
@@ -419,6 +555,12 @@ macro_rules! implement_page_frame_range {
                         $TypeName::new(start, end)
                     }
                 }
+            }
+            impl<P: PageSize + 'static + Copy> $TypeName<P> {
```

Comment:
is the `Copy` bound required here?

Reply:
The `Copy` bound is required, yes. I added the `'static' bound everywhere for consistencies sake, even where the compiler didn't request it. I only added the `Copy` bound where needed to avoid clutter, but I can add it everywhere for consistency just like `'static`, if that's preferred.

Reply:
Ok, but `Copy` _shouldn't_  be required. I think it's only necessary because of the way you implemented `AddAssign` and `SubAssign`, which is why I must not have implemented it that way in the past.

Where else (in what other methods) would `Copy` be specifically needed? In general we do not treat a range type as being `Copy`, since Rust doesn't (even though they easily could be).

Reply:
Using the original implementation of  `AddAssign` and `SubAssign` _does_ make it so that the `Copy` bound isn't needed for those traits.  
But removing the bound here results in an error message arising from a move from a shared reference.  `move occurs because value has type Page<P>, which does not implement the Copy trait`, for `*self.start()` and `*self.end()` in `overlap()`.

Reply:
Ah ok, right. In that case, no worries. 

I misinterpreted the bound as applying to the range type, but it applies to the `Page` type. So feel free to ignore this.

Only thing to change is that Rust typically places trait bounds before lifetime bounds, i.e., `P: PageSize + Copy + 'static`. Also I still don't see why `'static` is necessary, but we can look at that later.